src/anndata/_core/index.py

```diff
@@ -82,6 +82,10 @@ def name_idx(i):
             indexer = np.array(indexer)
             if len(indexer) == 0:
                 indexer = indexer.astype(int)
+        if isinstance(indexer, np.ndarray) and indexer.dtype == float:
```

Comment:
Please fix the comparison so it doesn’t only work with float64:

```pycon
>>> np.dtype(np.float64) == float
True
>>> np.dtype(np.float32) == float
False
```

(and parametrize the tests please)